book/src/images/overview_abstract.dot

```diff
@@ -0,0 +1,53 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+digraph {
+    graph [fontname = "sans-serif"; nodesep = 1;];
+    edge [fontname = "sans-serif";];
+    node [color = "#0077c8"; fontcolor = white; fontname = "sans-serif"; shape = box; style = "filled,rounded";];
+
+    // Define our nodes
+    qobject [group = left;label = "QObject subclasses\nQt-side representation";];
+    bridge [group = center;label = "CXX-Qt Bridge\nQt/Rust boundary description";];
+    rust [group = right;label = "Rust implementation";];
+
+    // Define our fake nodes for layout as splines=ortho doesn't suppoort ports
+    subgraph {
+        node [shape = point; style = invis; width = 0;];
+
+        left [group = left;];
+        center [group = center;];
+        right [group = right;];
+    }
+
+    // Define our ranks and specific order so nodes are in the correct positions
+    subgraph {
+        edge [style = invis;];
+
+        subgraph {
+            rank = same;
+            left -> center -> right;
+        }
+        subgraph {
+            rank = same;
+            qobject -> rust;
+        }
+    }
+
+    // Connect bridge to our fake nodes
+    subgraph {
+        edge [dir = none;];
+
+        bridge -> center;
+        center -> left [label = "Macro Expansion";];
+        center -> right [label = "Code Generation";];
```

Comment:
This is the wrong way round ;)

This does bring up the point though:
Currently we're the opposite of the CXX overview diagram (C++ on the left, Rust on the Right).
So I suggest flipping the nodes, instead of the arrow labels.
Can you move "Rust implementation" to the left and "QObject subclass" to the right?

Reply:
Sure :-)